enumerable_methods.rb

```diff
@@ -0,0 +1,65 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    my_each { |n| arr.push(n) if yield n } if block_given?
+    arr
+  end
+
+  def my_all?
+    my_select { |n| yield n }.length == length if block_given?
+  end
```

Comment:
- [ ] enumerables #my_all? returns false if the block returns false or nil::range
     Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block))